usr/src/uts/common/fs/zfs/spa.c

```diff
@@ -5471,6 +5472,7 @@ spa_vdev_remove(spa_t *spa, uint64_t guid, boolean_t unspare)
 		    ZPOOL_CONFIG_L2CACHE, l2cache, nl2cache, nv);
 		spa_load_l2cache(spa);
 		spa->spa_l2cache.sav_sync = B_TRUE;
+		spa_event_notify(spa, vd, ESC_ZFS_VDEV_REMOVE_AUX);
```

Comment:
I realize this was already merged, but it still seems like the best place to ask...

Won't `vd` always be NULL at this point?  At the beginning of this function, we do a `spa_lookup_by_guid` with `B_FALSE` for the aux argument.  This means that it'll return NULL for l2arc device guids, right?  If that's true, then when we get here, `vd == NULL`.  Would it be clearer to explicitly pass NULL here?